### ISSUE TITLE
0.6.5 PANIC at zfs_uio.c:187:uio_prefaultpages()

### ISSUE THREAD

**Issue by alexanderhaensch** [zfsonlinux/zfs#3792]
upgraded from 0.6.4.2 to 0.6.5 and recieved kernel panics occuring when  ESET Remote Administrator Software starts:

[  431.936604] PANIC at zfs_uio.c:187:uio_prefaultpages()
[  431.936686] Showing stack for process 20921
[  431.936688] CPU: 13 PID: 20921 Comm: RDSensor Tainted: P           O 3.14.51-hardened #1
[  431.936690] Hardware name: Supermicro X9DR3-F/X9DR3-F, BIOS 3.0a 07/31/2013
[  431.936691]  00000000000000bb ffff881f6c92b920 ffffffff815f25a5 ffffffffa07b4ce8
[  431.936694]  ffff881f6c92b930 ffffffffa076a0ef ffff881f6c92bab0 ffffffffa076a1a7
[  431.936696]  0000000000000246 ffff882000000030 ffff881f6c92bac0 ffff881f6c92ba60
[  431.936697] Call Trace:
[  431.936705]  [<ffffffff815f25a5>] dump_stack+0x45/0x56
[  431.936722]  [<ffffffffa07b4ce8>] ? zcommon_fini+0x1798/0x2fa8 [zcommon]
[  431.936728]  [<ffffffffa076a0ef>] spl_dumpstack+0x3f/0x50 [spl]
[  431.936732]  [<ffffffffa076a1a7>] spl_panic+0xa7/0xe0 [spl]
[  431.936736]  [<ffffffff8110b809>] ? __kmalloc_node+0x29/0x1a0
[  431.936739]  [<ffffffff815f68c9>] ? mutex_unlock+0x9/0x10
[  431.936760]  [<ffffffffa0a08cdc>] ? refcount_add_many+0x9c/0x140 [zfs]
[  431.936764]  [<ffffffffa07b28ec>] uio_prefaultpages+0x13c/0x150 [zcommon]
[  431.936777]  [<ffffffffa0a6f0ec>] zfs_write+0x1fc/0xe60 [zfs]
[  431.936779]  [<ffffffff810e3969>] ? zone_statistics+0x79/0x90
[  431.936782]  [<ffffffff810d20ab>] ? get_page_from_freelist+0x4cb/0x8e0
[  431.936791]  [<ffffffffa0aa5bef>] ? _fini+0xb156/0x7b0a3 [zfs]
[  431.936805]  [<ffffffffa0a08f34>] ? refcount_remove_many+0x194/0x290 [zfs]
[  431.936816]  [<ffffffffa0a8e209>] zpl_write_common_iovec.constprop.7+0x79/0x100 [zfs]
[  431.936826]  [<ffffffffa0a8e2fa>] zpl_aio_write+0x6a/0x90 [zfs]
[  431.936829]  [<ffffffff81118677>] do_sync_readv_writev+0x47/0x80
[  431.936831]  [<ffffffff81119db4>] do_readv_writev+0xc4/0x220
[  431.936841]  [<ffffffffa0a8e290>] ? zpl_write_common_iovec.constprop.7+0x100/0x100 [zfs]
[  431.936850]  [<ffffffffa0a8e320>] ? zpl_aio_write+0x90/0x90 [zfs]
[  431.936853]  [<ffffffff81119f8b>] vfs_writev+0x2b/0x50
[  431.936856]  [<ffffffff8111a0c5>] SyS_writev+0x45/0xd0
[  431.936858]  [<ffffffff815f897e>] system_call_fastpath+0x16/0x1b

**zpool status**
  pool: storage
 state: ONLINE
status: Some supported features are not enabled on the pool. The pool can
        still be used, but some features are unavailable.
action: Enable all features using 'zpool upgrade'. Once this is done,
        the pool may no longer be accessible by software that does not support
        the features. See zpool-features(5) for details.
  scan: scrub canceled on Tue Sep  8 00:25:16 2015
config:

        NAME                                                STATE     READ WRITE CKSUM
        storage                                             ONLINE       0     0     0
          raidz2-0                                          ONLINE       0     0     0
            sdn                                             ONLINE       0     0     0
            sdm                                             ONLINE       0     0     0
            sdl                                             ONLINE       0     0     0
            sdk                                             ONLINE       0     0     0
            sdj                                             ONLINE       0     0     0
            sdi                                             ONLINE       0     0     0
            sdt                                             ONLINE       0     0     0
            sds                                             ONLINE       0     0     0
            sdr                                             ONLINE       0     0     0
            sdq                                             ONLINE       0     0     0
            sdp                                             ONLINE       0     0     0
            sdo                                             ONLINE       0     0     0
            sdx                                             ONLINE       0     0     0
            sdw                                             ONLINE       0     0     0
        logs
          mirror-1                                          ONLINE       0     0     0
            sdd1                                            ONLINE       0     0     0
            sda1                                            ONLINE       0     0     0
        cache
          ata-INTEL_SSDSC2BB240G4_BTWL34210B2F240NGN-part1  ONLINE       0     0     0
          ata-INTEL_SSDSC2BB240G4_BTWL342203A9240NGN-part1  ONLINE       0     0     0
          ata-INTEL_SSDSC2BB240G4_BTWL342204AH240NGN-part1  ONLINE       0     0     0
          ata-INTEL_SSDSC2BB240G4_BTWL342203AM240NGN-part1  ONLINE       0     0     0
        spares
          d5                                                UNAVAIL 
          d6                                                UNAVAIL 

errors: No known data errors

**uname -a**
Linux io 3.14.51-hardened #1 SMP Thu Sep 17 12:21:09 CEST 2015 x86_64 Intel(R) Xeon(R) CPU E5-2620 v2 @ 2.10GHz GenuineIntel GNU/Linux

All ESET Software creates this error: RDSensor, ERAServer and ERAAgent.
These are proprietary Antivirus software tools. 
They do not function on the system correctly with 0.6.5

**Comment by ryao** [comment]
@alexanderhaensch #3813 should fix this.

**Comment by alexanderhaensch** [comment]
The github search did not reveal the other issues.. 
I can confirm, that the patch successfully solved the problem.

**Comment by alexanderhaensch** [comment]
Now i see this during a receive with parallel scrub:
Can it be related to the patch?
It is the only thing that changed.
[64345.893387] VERIFY3(mintxg == dsl_dataset_phys(ds1)->ds_prev_snap_txg) failed (6614665 == 6614099)
[64345.893391] PANIC at dsl_scan.c:983:dsl_scan_ds_clone_swapped()
[64345.893392] Showing stack for process 6430
[64345.893395] CPU: 6 PID: 6430 Comm: txg_sync Tainted: P           O 3.14.51-hardened #1
[64345.893396] Hardware name: Supermicro X9DR3-F/X9DR3-F, BIOS 3.0b 06/30/2014
[64345.893398]  00000000000003d7 ffff882024bdb918 ffffffff815f25a5 ffffffffa0ad68df
[64345.893401]  ffff882024bdb928 ffffffffa07790ef ffff882024bdbaa8 ffffffffa07791a7
[64345.893402]  ffff88201e672000 ffff882000000030 ffff882024bdbab8 ffff882024bdba58
[64345.893404] Call Trace:
[64345.893427]  [<ffffffff815f25a5>] dump_stack+0x45/0x56
[64345.893473]  [<ffffffffa0ad68df>] ? _fini+0x2ce36/0x7b2a3 [zfs]
[64345.893480]  [<ffffffffa07790ef>] spl_dumpstack+0x3f/0x50 [spl]
[64345.893486]  [<ffffffffa07791a7>] spl_panic+0xa7/0xe0 [spl]
[64345.893495]  [<ffffffffa09c1949>] ? dmu_buf_rele+0x9/0x10 [zfs]
[64345.893510]  [<ffffffffa0a561bd>] ? zap_unlockdir+0x5d/0x110 [zfs]
[64345.893524]  [<ffffffffa0a5639e>] ? zap_lookup_norm+0xbe/0x1b0 [zfs]
[64345.893537]  [<ffffffffa0a564be>] ? zap_lookup+0x2e/0x30 [zfs]
[64345.893551]  [<ffffffffa0a50fad>] ? zap_lookup_int_key+0x4d/0x60 [zfs]
[64345.893566]  [<ffffffffa0a09930>] dsl_scan_ds_clone_swapped+0xc0/0x470 [zfs]
[64345.893578]  [<ffffffffa09f0467>] dsl_dataset_clone_swap_sync_impl+0x757/0x830 [zfs]
[64345.893590]  [<ffffffffa09dc10e>] dmu_recv_end_sync+0xbe/0x400 [zfs]
[64345.893605]  [<ffffffffa0a0b59a>] dsl_sync_task_sync+0x12a/0x130 [zfs]
[64345.893618]  [<ffffffffa0a02b33>] dsl_pool_sync+0x493/0x5e0 [zfs]
[64345.893634]  [<ffffffffa0a24785>] spa_sync+0x365/0xde0 [zfs]
[64345.893649]  [<ffffffffa0a3946a>] txg_sync_thread+0x3ba/0x690 [zfs]
[64345.893665]  [<ffffffffa0a390b0>] ? txg_thread_wait+0xd0/0xd0 [zfs]
[64345.893670]  [<ffffffffa07751d5>] thread_generic_wrapper+0x75/0xb0 [spl]
[64345.893675]  [<ffffffffa0775160>] ? __thread_exit+0x20/0x20 [spl]
[64345.893681]  [<ffffffff81062614>] kthread+0xc4/0xe0
[64345.893683]  [<ffffffff81062550>] ? kthread_create_on_node+0x160/0x160
[64345.893687]  [<ffffffff815f88c9>] ret_from_fork+0x49/0x80
[64345.893689]  [<ffffffff81062550>] ? kthread_create_on_node+0x160/0x160

**Comment by ryao** [comment]
@alexanderhaensch That is a separate issue. Please file another issue so taht we can track it.

**Comment by behlendorf** [comment]
Resolved by  b815ec3  which will be cherry-picked in to 0.6.5.2 release.
